main.cpp

```diff
@@ -159,20 +204,51 @@ Thing 1) Hairdressing Salon
         bool qualifiedToColourHair = true;
         float haircutRate = 65.0f;
         std::string name = "Tracey";
+        int numberOfPersonalClients = 1;
 
         int getNumberOfPersonalClients();
         void cutCustomerHair();
         void takeBreak();
     };
 
-    // Cut Hair
     void cutHair(Hairdresser hairdresser);
-    // Colour Hair
     void colourHair(Hairdresser hairdresser);
-    // Wash Hair
     void washHair(Hairdresser hairdresser);
+};
+
+int HairdressingSalon::Hairdresser::getNumberOfPersonalClients()
+{
+    return numberOfPersonalClients;
+}
+
+void HairdressingSalon::Hairdresser::cutCustomerHair()
+{
+
+}
+        
+void HairdressingSalon::Hairdresser::takeBreak()
+{
+
+}
+
+void HairdressingSalon::cutHair(Hairdresser hairdresser)
+{
+    hairdresser.numberOfPersonalClients += 1;
+}
+    
+void HairdressingSalon::colourHair(Hairdresser hairdresser)
+{
+    hairdresser.numberOfPersonalClients += 1;
+}
+
+void HairdressingSalon::washHair(Hairdresser hairdresser)
+{
+    hairdresser.numberOfPersonalClients += 1;
+}
```

Comment:
This is a common issue with new programmers or programmers who are coming from garbage-collected languages (python, C#). 
When you pass this type into this function, the instance that is used inside this function is NOT the same instance that was passed in. 
That means any changes you make to the instance being used INSIDE the function are NOT kept when this function ends. 
the C++ concept of References and Pointers has not been covered at this point in the course.  If you wanted this to work as you are most likely expecting it to work, you would need to use references or pointers.
I don't let people use coding concepts in projects until after they've been covered in the course.
You don't need to revise these function parameters.
I just want you to be aware of the potential issue.

Reply:
I have finished the implementations. Can I get a review.